### PR TITLE
remove nose dependency in tests

### PR DESCRIPTION
**Pull request by kmaehashi**
`run_module` function still depends on `nose`.
I changed to use `pytest`.

### CODE REVIEW

File: chainer/testing/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import nose
+import pytest
```
Comment (niboshi) [inline chainer/testing/__init__.py]: This line will cause `ImportError` when importing `chainer.testing` in the environment without `pytest`. (pytest is no longer a requirement for normal chainer installation).
How about importing `pytest` in `run_module` function?

Comment (kmaehashi) [reply chainer/testing/__init__.py]: Fixed as suggested!

**Comment by kmaehashi** [comment]
Removed doctest from the decorator as `pytest` does not support interactive test run.
https://stackoverflow.com/questions/26840034/can-tests-with-pytest-fixtures-be-run-interactively

**Comment by takagi** [comment]
LGTM!

**Pull request merged**